File: old/javascript/dp-e03.js
const UPS = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
const LOWS = "abcdefghijklmnopqrstuvwxyz";

function shiftCharBy(char, shift) {
  let normchar = char.toLowerCase();
  let pos = (LOWS.indexOf(normchar) + shift) % LOWS.length;
  if (LOWS.indexOf(normchar) != -1) {
    return UPS.indexOf(char) > -1 ? UPS[pos] : LOWS[pos]
  } else {
    return char;
  }
}

function shiftStringBy(str, shift) {
  return str.split('').map((x) => shiftCharBy(x, shift)).join('');
}

let args = process.argv.slice(2);
console.log( shiftStringBy(args[0], parseInt(args[1])) );

File: old/python/dp-e03.py
from string import ascii_uppercase as ups, ascii_lowercase as lows

def shift_char_by(char, shift):
    # access ascii_uppercase array via char position + shift, wrapped for overflows.
    try:
        pos = (ups.index(char.upper()) + shift) % len(ups)
    except: # if an error is raised, return char (cos its not a letter)
        return char

    # return from appropriate array.
    return ups[pos] if char in ups else lows[pos]

def shift_string_by(strng, shift):
    shifted = [shift_char_by(x, shift) for x in list(strng)]
    return ''.join(shifted)

inp = raw_input("string >>    ")
shift = int(raw_input("shift by >>    "))
print(shift_string_by(inp, shift))
